src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -469,13 +507,52 @@ void WiFiManager::ConnectHandler(Platform::UniquePtr<uint8_t> data)
     }
 }
 
-void WiFiManager::DisconnectHandler(Platform::UniquePtr<uint8_t>)
+void WiFiManager::DisconnectHandler(Platform::UniquePtr<uint8_t> data)
 {
-    SystemLayer().ScheduleLambda([] {
+    CHIP_ERROR err = SystemLayer().ScheduleLambda([capturedData = data.get()] {
+        Platform::UniquePtr<uint8_t> safePtr(capturedData);
+        uint8_t * rawData          = safePtr.get();
+        const wifi_status * status = reinterpret_cast<const wifi_status *>(rawData);
```

Comment:
This seems very fragile, in that we have no checks to make sure the data size is correct, etc..... And nothing even documents what the expected data type here is.

As a followup: if the input to this stuff were at least ScopedMemoryBufferWithSize then you could check the sizing up front (outside the lambda)....

Reply:
Fair point, thanks for the catch. I passed data size to the handlers API and added checks validating if size matches the expectation.